### ISSUE TITLE
Wrong version is tagged `latest`

### ISSUE THREAD

**Issue by wraithgar** [JoshGlazebrook/socks#95]
Currently `socks@2.7.3` is tagged as `latest` in npm, which means folks installing it for the first time, or running `npm update` will get that version instead of `socks@2.8.0`

What I would suspect happened is that `2.7.2` and `2.7.3` were uploaded after `2.8.0` without specifying an alternate tag, and were thus tagged `latest` by default.

To fix this, maintainers can run `npm dist-tag add socks@2.8.0 latest`

**Comment by wraithgar** [comment]
There does not appear to be any functional difference between the two versions, except for a change in the `engines` field (which arguably is a breaking change), and the removal of the `build-raw` script.

Folks who are seeing this can probably ignore it till it's either fixed, or a new `socks` version is published and tagged as `latest`.

```
~/D/n/c/b/gar_deps-updates (gar/deps-updates|✚2) $ npm diff --diff socks@2.7.3 --diff socks@2.8.0
diff --git a/package.json b/package.json
index v2.7.3..v2.8.0 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "socks",
   "private": false,
-  "version": "2.7.3",
+  "version": "2.8.0",
   "description": "Fully featured SOCKS proxy client supporting SOCKSv4, SOCKSv4a, and SOCKSv5. Includes Bind and Associate functionality.",
   "main": "build/index.js",
   "typings": "typings/index.d.ts",
@@ -23,7 +23,7 @@
     "socks5"
   ],
   "engines": {
-    "node": ">= 10.0.0",
+    "node": ">= 16.0.0",
     "npm": ">= 3.0.0"
   },
   "author": "Josh Glazebrook",
@@ -52,7 +52,6 @@
     "test": "NODE_ENV=test mocha --recursive --require ts-node/register test/**/*.ts",
     "prettier": "prettier --write ./src/**/*.ts --config .prettierrc.yaml",
     "lint": "eslint 'src/**/*.ts'",
-    "build": "rm -rf build typings && prettier --write ./src/**/*.ts --config .prettierrc.yaml && tsc -p .",
-    "build-raw": "rm -rf build typings && tsc -p ."
+    "build": "rm -rf build typings && prettier --write ./src/**/*.ts --config .prettierrc.yaml && tsc -p ."
   }
 }
```

**Comment by MartinFalatic** [comment]
Why does the 2.7.3 tagged package-lock file show this as v3.0.0 now?

https://github.com/JoshGlazebrook/socks/blob/2.7.3/package-lock.json

https://github.com/JoshGlazebrook/socks/compare/2.7.1...2.7.3

I've had a devil of a time trying to get this to upgrade as part of `socks-proxy-agent@8.0.2` (it refuses for no obvious reason to go past socks 2.7.1 despite specifying the dependency as ^2.7.1) and I think this is why.

https://github.com/TooTallNate/proxy-agents/issues/293#issuecomment-1960652215

**Comment by JoshGlazebrook** [comment]
Okay I've published socks@2.7.4 and I also published socks@2.8.1 both with correct package-lock.json

I think this should fix the semver upgrades for the 2.7.x versions, and 2.8.1 should fix it for the 2.8.0 that was published under a beta tag.

**Comment by wraithgar** [comment]
```
~ $ npm view socks version
2.8.1
~ $ npm dist-tag ls socks
beta: 2.7.2
latest: 2.8.1
```
